### ISSUE TITLE
Steam build update

### ISSUE THREAD

**Issue by kikotheexile** [endless-sky/endless-sky#3637]
This is more of a public relations issue than a mechanical one, although mechanics are also in question here. I couldn't think of a reasonable location to share this, though. 

I fully understand that there are many reasons why the steam build has not been updated in a while, it would be nice to have an update so that it is in line with some of the more recent changes made to the github version. Less as a "here, have a ton of new content!" release and more of a "here are some of the improvements that have been made!" update. Something less 0.9.9 and more 0.9.8(b) just to let those on steam know that this isn't abandonware and that it is, indeed, still being worked on. 

Not every update to the steam build needs to have massive content releases. Even simple enhancement updates go a long way, and the game has made some pretty reasonable improvements in the past 6 months.

**Comment by jafdy** [comment]
There have been "Is this abandoned" posts coming up from about 2 month after a release. We do explain that development is quite slow but having more regular (if tiny) updates would help.
If you really want to have big content addition updates then you can just not include the major sections of content until a major update.

**Comment by ghoulavenger** [comment]
Since I currently play the Steam release myself I wouldn't mind if there were more steam updates, even if they were small ones. I think the github would be better off for experimentation and incomplete story arcs. 

That being said, I don't see a real issue with delaying steam releases since you CAN just download and compile the game yourself. Granted, it isn't exactly easy to get it to compile in windows -- I had a lot of problems with CodeBlocks or whatever it's called, never could find the right libraries, so I eventually gave up (not worth the hassle).

**Comment by KiLEdEnNis** [comment]
> That being said, I don't see a real issue with delaying steam releases since you CAN just download and compile the game yourself.

1 in 100 would try, 1 in 500 would succeed (example numbers)
what about the other 499

smaller updates every 2, 3 months wouldn't hurt much imo

**Comment by ghoulavenger** [comment]
> 1 in 100 would try, 1 in 500 would succeed (example numbers)
what about the other 499

The people that really want it will find a way to do it, which was my point. The rest should at least have the capacity to google the github and find out the game is being actively worked on. And if you read earlier in my post -- I don't object to more frequent steam updates, it just isn't a huge issue to me.

**Comment by KiLEdEnNis** [comment]
> The people that really want it will find a way to do it, which was my point.

> I had a lot of problems with CodeBlocks or whatever it's called, never could find the right libraries, so I eventually gave up

it's hard to do and shouldn't require this much effort for everyone
also remember a part of the playerbase may be young children, and a part that's... less smart etc.
if ya really want it it doesn't mean ya can do it

there's no harm in some periodic updates with smaller fixes

**Comment by Kryes-Omega** [comment]
I think that Kile is right in this case. Not everyone has the tech know-how to compile games. Not everyone knows there is a git repository. Some people just like the game and want to press 'Play' and have a good time. This shouldn't be something where we force average players to figure out how to compile the program just so they or their kids can play, "if they really want it," right?

**Comment by Tadrix** [comment]
I doubt this is a good idea. If it was optional, I'd be all for it, but Steam has a habit to automatically update games as soon as there's an update. AFAIK, for now Steam versions are at least to some degree tested (even if not all of them are stable releases), so there aren't any game-breaking bugs. If we'd just dump entire content every, say, 2 of 3 months, then we would be treating our player base as testers (and testing in production, however popular these days, isn't desirable). More work on intermediate releases would detract from generation of new content (and development would be even "slower").

**Comment by Brick63** [comment]
there is such a thing as an unstable release to then be made stable if it has no huge bugs in it

**Comment by tehhowch** [comment]
The only sane way to simply prep smaller Steam releases that hold back in-progress things would be to have multiple branches. This would not complicate things too much (multiple branches were used in the early days), but could make accessing the latest content more difficult for those who are not familiar with GitHub.

I'm not against this, but MZ doesn't generally have enough burners lit at once to justify needing multiple branches to organize feature development.

**Comment by MCOfficer** [comment]
i guess this is the correct time to advertise my own efforts in this direction: [http://steamcommunity.com/app/404410/discussions/0/1700542332319963311/](the nightly builds forumpost)

**Comment by kikotheexile** [comment]
There is a beta branch...

**Comment by jafdy** [comment]
Just sticking whatever we have on the beta branch every 3-4 months would be plenty I expect. In this case it's not about keeping people trying to contribute up-to-date and more about giving the steamies new stuff to play with from time to time.

**Comment by Wrzlprnft** [comment]
and help woth qa for the masses

**Comment by Amazinite** [comment]
Since v0.9.1 (when the unstable/stable release cycle began), the average time between stable releases has been 141.25 days and the average time between unstable releases has been 140 days. The time between v0.9.8 and now has been 178 days, and the time between v0.9.7 and now has been 203 days. Time between a stable release and the unstable release that comes after it seems to have been increasing every update as well:

Release | Days 
--------|------
v0.9.2 to .3 | 81
v0.9.4 to .5 | 105
v0.9.6 to .7 | 149
v0.9.8 to today | 178

While the amount of content per unstable update has also been increasing (which is great), the increased time between large content dumps has lead to an increase in people asking on Steam if development is still in progress. That isn't to say that updates should be rushed, and I for one am fine with the current system, but releasing more frequent updates (say 3 months ± 2 weeks max between updates), even if the updates are smaller in size, does go a long way to keeping players happy and interested in the game.

If you (MZ) wish to keep the v0.9.x where x = odd being unstable and x = even is stable system, then we could tag on an extra number for mini updates (e.g. v0.9.9.1 [unstable] could be released today and v0.9.9.2 [unstable] could be released when the Wanderer campaign is done).

**Comment by gunqqer** [comment]
I do support having more smaller updates as someone who can't currently compile it myself (I'm running linux and am having some library version conflicts with SDL2 after doing some updates.) Having updates would be very nice even if they are small

**Comment by endless-sky** [comment]
The fact is, I don't have a Windows build computer set up at the moment, and I on;y just recently got a replacement for my 7-year-old Mac that bit the dust. That's part of why there haven't been any new releases.

Also, I was hoping for 0.9.9 to wrap up the Wanderer story line, but other things keep getting in the way...

**Comment by kikotheexile** [comment]
@endless-sky may i suggest making Derpy Horse (Amazinite) a "developer" on Steam for the sake of updating the steam releases and store page?  

Would make community management and growth a lot easier on you and would be one less thing to eat your time.

**Comment by Amazinite** [comment]
Kiko's suggestion is his own, but I'd be fine with having more tools for community management on Steam. I'm unsure if there's some sort of administrator role between developer and moderator on the Steam forums, but the moderator role does provide a lack of tools to do true community management with. 

Discussing with some others and without knowing all that I would be able to do, my first thoughts were updating the different subforum sections of the discussion forums (e.g. making the mods subforum not a trading forum so that posts don't disappear and possibly adding more subforums with different topics such as a "Content Creation" subforum similar to that which is on the Google Group but separate from the subforum used to house working plugins) and updating the video/screenshots on the store page to be up-to-date and more interesting in order to make a better first impression of the game beyond the existing and in my opinion rather bland 1 minute 20 seconds of random gameplay of a Leviathan beating up some pirate interceptors and small warships.

**Comment by kikotheexile** [comment]
One of the reasons i suggest @Amazinite for developer on steam is because he is a spreadsheet nerd and can/does already provide tons of feedback on how to best create mods and additions to the game, among other things. With more tools, he could better direct the community and provide an answer with authority when people ask questions to the developer. 

Basically all the way around, he is perfectly suited for the job and simply just needs the tools to execute it.

**Comment by arkhne** [comment]
I signed up to say this. As a steam user, and barely a dabbler in programming (my knowledge is old, and limited), I've been coming here regularly to check on progress, but have not tried compiling, as I don't feel that invested in re-learning. I don't particularly mind/care when additional content (such as new stories) is released. What I DO care about (And what I am mostly awaiting) is Quality of Life improvements. Like the zoom increments, which is a MASSIVE quality of life improvement for those of us who's resolutions don't perfectly match the devs.

I do understand the desire to run the un/stable pattern, however, when you are gating Quality of Life improvements that are already done, behind time-consuming additional/bonus content, it does not feel good as the end user.

I also totally understand the technical limitations of broken/old machines, and I am not pushing for an update, I just wanted to add the QoL perspective.

(I also don't want to sound ungrateful, this is one of my favourite games out there, so I do definitely appreciate the work, and don't want to come across incorrectly).

**Comment by MCOfficer** [comment]
i just caught myself looking up obituaries oft your hometown. sure as hell can't blame you if anything serious happened, but if not... dang, a lifesign would be nice

**Comment by Amazinite** [comment]
I've gotten a few messages from random people on Steam as well. Instead of just updaing now, I think a PSA news post on Steam as to why the update is taking a while would help.

**Comment by kikotheexile** [comment]
And unfortunately the only person who can has a life and no help that can actually cover for them.

**Comment by jafdy** [comment]
If the problem is a windows computer then you could just grab some hard drive (with 40GB or so of space) and boot a copy of windows 10 onto it using any other computer. The download doesn't require a key and the activation (with a key) can be delayed for a while. That would allow you to do a windows build. 
It's a heck of a work around but it would do it.

**Comment by KiLEdEnNis** [comment]
if it's what jafdy said, I could hook you up with Windows 10 Education from Dreamspark (free software for people that are learning IT, like me in college)
![image](https://user-images.githubusercontent.com/10348657/38175659-59423aca-35e0-11e8-8c78-137a78f04a62.png)

if you need it, all you have to do is ask

**Comment by jafdy** [comment]
You can just get installation media from the microsoft website here https://www.microsoft.com/en-gb/software-download/windows10 (or local equivalent).

**Comment by endless-sky** [comment]
My company is currently sitting on a pile of approximately 50 very capable old Windows laptops that they're promising to sell back to the employees for cheap, one of these days. (We were supposed to get access to them a month ago, actually.) So, I could either go with that or just toss Windows on a small hard drive and add it to my Linux computer. So, if I had an urgent need to a Windows build machine, I could track one down.

If I have all three computers sitting side by side, the actual build process takes less than an hour; most of the time involved in a release is writing up the release notes, updating the website, adding a new post on Steam, etc. But in general, doing builds is a relatively tiny portion of the development time. (And, doing releases more frequently wouldn't make sense because a certain amount of polishing and testing work needs to be done before each release - it wouldn't work to just push out an extra build mid-cycle when things are still in flux.

**Comment by FireWolf74** [comment]
Windslow RUNNNNNNNNN,

**Comment by KiLEdEnNis** [comment]
what are you doing? take that shit to somewhere that ain't GitHub tyvm

**Comment by FireWolf74** [comment]
Wow guys talking about talk about lightening the mood.  Damn yall so up tight right now live a little.
https://www.youtube.com/watch?v=y6Sxv-sUYtM

**Comment by MCOfficer** [comment]
the point is that github is meant to be productive. this is not the place for messing with us or irrelevant banter.
all you're doing is polluting a conversation, hence the backlash.

**Comment by FireWolf74** [comment]
MCOfficer,
I was doing a reaction to the installing of (cringeworthy) Windows 10.  While I have spent many of days helping people get off Windows because of the Data Corruption that has been happening in recent months, I have had to rebuild 6 machines so far the last 4 months sometime seeing Windows and other ilk makes me want to scream RUNN! And if you never had to rebuild data from a level of corruption that would make certain Government Agencies happy then I guess you will never have the hate for an OS like I have.

**Comment by jafdy** [comment]
According to the steam survey, 96.3% of steam users use some form of windows. Of those, 57.66% (55.53% of total) are using 64bit Windows 10 and 34.84% (33.53% of total) are using 64bit Windows 7. OSX commands most of the remaining share with Linux covering only 0.57%, much less than 32bit Windows 7. 

Why do I point this out? Although this may not be perfectly representative of the computers of the world, even with humongous margins of error, Windows is by far the most popular OS and it would be crazy to stop development for Windows and cut out that many people. 

Regardless of what issues you may personally have with Windows development will be continuing. 
Source: https://store.steampowered.com/hwsurvey/Steam-Hardware-Software-Survey-Welcome-to-Steam

@MCOfficer, I'm dropping it now, I couldn't resist a good pile of stats to throw at someone.

**Comment by asl97** [comment]
> Windows is by far the most popular OS

Most popular? No way, it's just the most commonly used. (there is a different)
____
Anyway, global stats are pretty pointless in my view.

If you want to throw some stats around, choose something better like project specific stats:
https://api.github.com/repos/endless-sky/endless-sky/releases/7390079
and https://launchpad.net/~mzahniser/+archive/ubuntu/endless-sky/+builds?build_state=built using https://gist.github.com/stefansundin/f9df6c5e0fd184c60709

File name | Download count
------------- | ----------------------
endless-sky-macosx-0.9.8.dmg | 2769
endless-sky-win32-0.9.8.zip | 1971
endless-sky-win64-0.9.8.zip | 7364
0.9.8-0zesty amd64 | 59
0.9.8-0zesty i386 | 6
0.9.8-0xenial amd64 | 668
0.9.8-0trusty i386 | 22
0.9.8-0trusty amd64 | 88
0.9.8-0xenial i386 | 41

Opt-in stats
https://qa.debian.org/popcon.php?package=endless-sky
https://popcon.ubuntu.com/by_inst

Other sources | count
------------------- | --------
debian popcon | 324
ubuntu popcon | 73

As for other linux distro users, there isn't any specific stats for them since there isn't any stats collection but one can probably take the git clone/fetch count spike after release as a base,  however such stats are only available to member with push access and they are only available for 14 days.

https://help.github.com/articles/viewing-traffic-to-a-repository/
____
In any case @jafdy points still stands, from the current publicly available stats, 64bit windows is at the top, then macs, follow by 32bit windows, finally linux.
____
ps: Download count for older release is also available through the api: https://api.github.com/repos/endless-sky/endless-sky/releases

**Comment by kikotheexile** [comment]
"You're wrong and here's why you're right"

Let's chill on the off topic semantics in reference to a half a year old thread last commented on two months ago and hope something happens

**Comment by KiLEdEnNis** [comment]
*receives an email from the guy he downvoted but can't be bothered to read it*

**Comment by Wrzlprnft** [comment]
Disclaimer: Behaviour as shown by KiLEdEnNis above is not what the part of the community i know is represented by or wants to encourage.

**Comment by asl97** [comment]
@KiLEdEnNis The issue was closed and @kikotheexile has stated to "chill".
I noticed your :-1: and merely asked privately by email so has not to cause more noise here.

> can't be bothered to read it

I am actually relieve that the -1 was due to laziness than an actual problem with my post :smile: 

If there was an incorrect information in my comment, I be glad to correct it.

But since you brought it up here again,  let me take this chance to explain my :-1: on @jafdy comment in case it wasn't clear:

Let's start with this quote:
> "it would be crazy to stop development for Windows and cut out that many people."

It would only be crazy if there was many windows endless-sky users.

The prove that jafdy provided was the global stats which jafdy claimed are and I quote: "good pile of stats".

I strongly disagree with that claim of it being "good" as it doesn't actually reflect the actual user base and it was my reason for my :-1:.

It's like claiming most users are from asia just cause the peak traffic is the highest (as of writing) https://store.steampowered.com/stats/content/ , It makes no sense and there nothing to prove it.

Not to mention, [the link](https://store.steampowered.com/hwsurvey/Steam-Hardware-Software-Survey-Welcome-to-Steam) jafdy gave state 'Participation in the survey is optional'.

That means users can choose not to participate, it may be bias of me to say this but I believe linux users are more likely to opt-out of such a thing while most windows users probably aren't even aware of it.

In any case, as those global stats doesn't reflect actual endless-sky users,
(they never did, they don't even include actual numbers, just percentage)

I researched and commented with actual stats with actual numbers that anyone can verify by themselves.

I provide what I believe is solid evidence that verify and backed up the claim "it would be crazy to stop development for Windows" that @jafdy made.
____

While my download count stats isn't good at getting a count of the active user base, it's at least shows something about the endless-sky user base OS, unlike global stats which doesn't show anything specifically about endless-sky.
____

In any case, my recent further research has found this other site with some interesting information https://steamdb.info/app/404410/graphs/ , unfortunately, that site doesn't provide OS information though.

Different people will have different view about the information, like maybe some would take it has it's highly unlikely for the global stats percentage to reflect the actual active user base as it's less than 0.000001 of the total steam user base. (~100 out of 10000000+)

There isn't any active user OS stat available anywhere publicly as far as I am aware of.

PS: 7 minutes average playtime in last 2 weeks is kind of depressing :disappointed: 
Not to mention the fact that most people don't play more than half-hour total time.

**Comment by kikotheexile** [comment]
Due to DirectX, the vast majority of gamers or users in general use Windows based technology. Although Linux and Mac based platforms can emulate DirectX, Emulation is not the same as running it in its native environment. 

This is why Steam Machines aren't currently the dominant form of computers used by Steam users. The operating system designed for gaming is incompatible with most games, so its really no surprise that most people use the operating system that is most likely to run a game out of the box. I mean, after all, Microsoft made their way into the console market with their "X-Box" which was 100% just taking DirectX and making a stand alone platform with it. 

Your argument is not only wrong, its off topic, and not really suited for Github let alone this issue. Github for this game is usually kept to the development of this game, not to conversations about different operating systems and their popularity. We have a discord for this sort of thing where people would be much more willing to discuss this in greater detail, with less cold and at-arms-length statements.
